app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -244,11 +246,12 @@ class BrowserTabViewModel(
         autoCompleteViewState.value = AutoCompleteViewState(false)
     }
 
-    override fun progressChanged(newProgress: Int) {
+    override fun progressChanged(newProgress: Int, canGoBack: Boolean, canGoForward: Boolean) {
         Timber.v("Loading in progress $newProgress")
 
         val progress = currentLoadingViewState()
         loadingViewState.value = progress.copy(progress = newProgress)
+        browserViewState.value = currentBrowserViewState().copy(canGoBack = canGoBack, canGoForward = canGoForward)
```

Comment:
Optional: It feels like the browserViewState and loadingViewState should be merged as they closely coupled and we are updating both anyway.

Reply:
It's tricky to draw the lines between them, as they do all feel pretty related.

However, I'd leave those two separate because they don't always both get updated. While a change in progress here does result in changes to both, there are other places where a change to the browser state **does not** result in changes to progress. So they are, at times, completely independent.